clang/tools/libclang/CDependencies.cpp

```diff
@@ -148,7 +154,7 @@ class FullDependencyConsumer : public DependencyConsumer {
 static CXFileDependencies *getFullDependencies(
     DependencyScanningWorker *Worker, ArrayRef<std::string> Compilation,
     const char *WorkingDirectory, CXModuleDiscoveredCallback *MDC,
-    void *Context, CXString *error) {
+    void *Context, CXString *error, ModuleDependencyCommandLine ModDepCmd) {
```

Comment:
I think this might be easier to follow / more direct if it took an `llvm::function_ref<CXStringSet *(const ModuleDeps &)>` and we dropped the `ModuleDependencyCommandLine` enum.

Or are there follow ups that will add more configuration points, where you think the enum might be cleaner?

Reply:
No particular reason to use enum, callbacks are fine too. Changed in the new revision.